### DISCUSSION TITLE
UGreen 2.5 Gbps USB-A ethernet adapter and DS918+

### DISCUSSION THREAD

**Discussion by leeivan8686** [bb-qq/r8152#435] [Q&A] [answered]
Asking for help on the ugreen adapter 2.5g usb-a version. Has it been reported it does not work with these drivers? I understand it uses the rtl8156bg chip just like the usb-c version. However I could not get it to be detected by DSM after retrying multiple times and even rebooting. My USB port is working and I have followed the instructions in the guide carefully. I am running the latest DSM version of 7.2.2 if it helps.

**Comment by bb-qq** [comment]
That product should also work.

 Please create an Issue and follow the template to submit your logs and I will investigate why it is not working.

**Comment by matthys70** [comment, marked as answer]
Just to let you know I got DS916+ (DSM 7.2.2-72806) with UGREEN USB-A to RJ45 2.5G Ethernet Adapter (Model CM648 P/N 25051) and is working. I did need to do the extra shell steps and afterward run the package in the Package Center (I hope this is a one time only) and has now the Installed status. Under network I now got LAN 3 and in the Info Center under Network Status it says: 2500 Mbps, Full duplex, MTU 9000 (manually set the MTU to 9000).
